pkg/mount/mount_unix_test.go

```diff
@@ -79,6 +81,58 @@ func TestMounted(t *testing.T) {
 	}
 }
 
+func TestMountTmpfsOptions(t *testing.T) {
+	if os.Getuid() != 0 {
+		t.Skip("root required")
+	}
+
+	testCases := []struct {
+		opts       string
+		expected   string
+		unexpected string
+	}{
+		{
+			opts:       "exec",
+			unexpected: "noexec",
+		},
+		{
+			opts:     "noexec",
+			expected: "noexec",
+		},
+	}
+
+	target := path.Join(os.TempDir(), "mount-tmpfs-tests-"+t.Name())
+	if err := os.MkdirAll(target, 0777); err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(target)
+
+	for _, tc := range testCases {
+		t.Run(fmt.Sprintf("%s:%q-%q-%q", t.Name(), tc.opts, tc.expected, tc.unexpected), func(t *testing.T) {
+			if err := Mount("tmpfs", target, "tmpfs", tc.opts); err != nil {
+				t.Fatal(err)
+			}
+			defer ensureUnmount(t, target)
+
+			mounts, err := GetMounts(nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			for _, entry := range mounts {
+				if entry.Mountpoint == target {
+					if !strings.Contains(entry.Opts, tc.expected) {
```

Comment:
This is not very important here, but the options are comma-separated list of entries and should be treated as such, so substring search is ambiguous here.

I'd suggest having something like
```go
opts := "," + entry.Opts + ","
if !strings.Contains(opts, ","+tc.expected+",") {
...
```
or something like this. This way you can add "exec" to "noexpected" for the second test case.

Reply:
Thanks, yes this will allow to add the `exec` option to `unexpected` field. Updated the PR.